### PR TITLE
NIAD-1154: Add TF deployment code for gpc-consumer adapter in Jenkins

### PR DESCRIPTION
**Pull request by RIashfaqahmed**
Add TF deployment code for gpc-consumer adapter in Jenkins

### CODE REVIEW

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
I don't think this belongs on the gpc repo, will catch up with you on it.

File: Jenkinsfile [hunk 1]
```diff
@@ -69,11 +79,43 @@ pipeline {
...
+                            if (publishWiremockImage) {
+                                if (sh (label: "Pushing Wiremock image", script: dockerPushCommandWiremock, returnStatus: true) !=0) { error("Docker push gpc-consumer-wiremock image failed") }
+                            }
                         }
                     }
                 }
             }
         }
+    
+        // stage('Deploy & Test') {
```
Comment (LukaszKwasniewski) [inline Jenkinsfile:93]: as we are not going to deploy gpc with this pipeline, this section is not needed and should be removed

File: Jenkinsfile [hunk 2]
```diff
@@ -89,3 +131,116 @@ int ecrLogin(String aws_region) {
     String dockerLogin = sh (label: "Getting Docker login from ECR", script: ecrCommand, returnStdout: true).replace("-e none","") // some parameters that AWS provides and docker does not recognize
     return sh(label: "Logging in with Docker", script: dockerLogin, returnStatus: true)
 }
+
+
+String tfEnv(String tfEnvRepo="https://github.com/tfutils/tfenv.git", String tfEnvPath="~/.tfenv") {
```
Comment (LukaszKwasniewski) [inline Jenkinsfile:136]: we are not going to deploy terraform with this pipeline, so the this function and the others below are not needed, please remove them.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,3 +1,8 @@
+String tfProject      = "nia"
+String tfEnvironment  = "build1"
+String tfComponent    = "gpc-consumer"
+Boolean publishWiremockImage = false // true: To publish gpc-consumer wiremock image to AWS ECR gp2gp-wiremock
```
Comment (LukaszKwasniewski) [inline Jenkinsfile:4]: on what conditions we should build and publish gpc wiremock?

Comment (RIashfaqahmed) [reply Jenkinsfile:4]: this has not been clarified yet but I found info about wiremock in Environment variables section (https://gpitbjss.atlassian.net/browse/NIAD-1154) . So I have added up a control to build wiremock image.

**Comment by RIashfaqahmed** [comment]
Devs confirmed no need of gpc-consumer wiremock image and no requirement of gpc-consumer adapter deployment in AWS via gpc-consumer repo (this would be done through Gp2GP repo). No more changes are required so closing this PR.

**Pull request closed without merging**